### PR TITLE
AP Ocarina of Time Client

### PR DESCRIPTION
**Pull request by espeon65536**
If this is not in time for the release that's fine, figured I'd just open this anyway. This is built in a very similar way to the FF1 client, but reading OoT data from Bizhawk instead. I've tested it fairly extensively in both solo games and public multiworlds, and have observed no issues. 

For now, I have not made any changes to the OoT ROM generation in order to continue supporting Z5Client. However, this may change in the future as updates to the randomizer cause changes to the memory layout.

### CODE REVIEW

File: OoTClient.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+        if payload['isDead']: # link is dead
+            ctx.deathlink_pending = False
+            if not ctx.deathlink_sent_this_death:
+                ctx.deathlink_sent_this_death = True
+                await ctx.send_death()
+        else: # link is alive
+            ctx.deathlink_sent_this_death = False
+
+
+async def n64_sync_task(ctx: OoTContext):
```
Comment (Berserker66) [inline OoTClient.py:143]: this task doesn't seem to call `await asyncio.sleep()`, while not connected to n64. This is probably a bad idea as we just waste CPU cycles checking as fast as possible.

File: OoTClient.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    Utils.init_logging("OoTClient")
+
+    async def main():
+        multiprocessing.freeze_support()
+        parser = get_base_parser()
+        parser.add_argument('apz5_file', default="", type=str, nargs="?",
+                            help='Path to an APZ5 file')
+        args = parser.parse_args()
+
+        if args.apz5_file:
```
Comment (Berserker66) [inline OoTClient.py:235]: Patching OoT can take a bit of time, right? Might be better to launch this as a task, so the client UI can load and init alongside.

**Review by Berserker66 — CHANGES_REQUESTED** [review]

File: OoTClient.py (hunk 1 above)
Comment (espeon65536) [reply OoTClient.py:143]: Line 202 calls `asyncio.wait_for()` for 10 seconds, so it doesn't repeat checking.

File: OoTClient.py (hunk 2 above)
Comment (espeon65536) [reply OoTClient.py:235]: Makes sense, will segment patching into its own task.

**Pull request merged**